adapters/IronSource/IronSourceAdapter/GADMAdapterIronSource.m

```diff
@@ -0,0 +1,204 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterIronSource.h"
+
+@interface GADMAdapterIronSource () {
+    
+    //Connector from Google Mobile Ads SDK to receive interstitial ad configurations.
+    __weak id<GADMAdNetworkConnector> _interstitialConnector;
+}
+
+@end
+
+@implementation GADMAdapterIronSource
+
+#pragma mark Admob GADMAdNetworkConnector
+    
+- (instancetype)initWithGADMAdNetworkConnector:(id<GADMAdNetworkConnector>)connector {
+    if (!connector) {
+        return nil;
+    }
+    self = [super init];
+    if (self) {
+        _interstitialConnector = connector;
+    }
+    return self;
+}
+
+- (void)getInterstitial {
+    id<GADMAdNetworkConnector> strongConnector = _interstitialConnector;
+    NSDictionary *credentials = [strongConnector credentials];
+
+    /* Parse enabling testing mode key */
+    GADMIronSourceExtras *extras = [strongConnector networkExtras];
+    self.isTestEnabled = extras.debugEnabled;
```

Comment:
The `GADMAdNetworkConnector`'s [`testMode`](https://developers.google.com/mobile-ads-sdk/docs/dfp/ios/api/reference/Protocols/GADMediationAdRequest#/c:objc(pl)GADMediationAdRequest(im)testMode) can be used to determine if a publisher is requesting test ads.